### PR TITLE
openjpeg: don't use vendored libs in tests

### PR DESCRIPTION
**Pull request by paparodeo**
the tests are re-building everything but not taking the build flags into consideration and using vendored libs which silently fail to build with clang-19 and result in failed tests due to missing binaries. Disabling 3rd party libs fixes the tests.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

@emilazy

### CODE REVIEW

**Comment by paparodeo** [comment]
> (Going to try to spend a little time to see if it’s possible to just make CTest work normally before merging this.)

I'm not supper confident that the test are actually passing. it seems that there is a required data repo we are not downloading: https://github.com/uclouvain/openjpeg-data.

downloading the data and then adding
```
    "-DJPYLYZER_EXECUTABLE=${lib.getExe' jpylyzer "jpylyzer"}"
    "-DOPJ_DATA_ROOT=./data"
```
to the `cmakeFiles` has  `ctest` emit less errors (the data files can be located) but still 10% of the tests fail.


building a "successful" test run then looking in the `Testing` directory shows lots of errors due to not finding the test data directory and I assume that the actual result is only 35% of the tests pass -- which is what just running `ctest` in the build directory shows.

**Comment by paparodeo** [comment]
> Aha. So the primary service the Travis CI CMake file is providing to us is to swallow all the errors?

it's weird -- like when it couldn't compile stuff due to the vendored zlib it didn't report the compile error, ran the tests, and then failed -- which is what this PR aimed to fix ( and does).

but, yeah, it seems like the test failures aren't getting reported. feel like I should've just left the tests disabled. they already were disabled for aarch64 and only were running on x64 darwin and linux.

even with 10% failing that's like 140 tests that would need to be disabled. maybe there is a regex that matches...

File: pkgs/by-name/op/openjpeg/package.nix [hunk 1]
```diff
@@ -39,11 +48,13 @@ stdenv.mkDerivation rec {
...
-    (mkFlag jpipLibSupport "BUILD_JPIP")
-    (mkFlag jpipServerSupport "BUILD_JPIP_SERVER")
+    (lib.cmakeBool "BUILD_JPIP" jpipLibSupport)
+    (lib.cmakeBool "BUILD_JPIP_SERVER" jpipServerSupport)
     "-DBUILD_VIEWER=OFF"
     "-DBUILD_JAVA=OFF"
-    (mkFlag doCheck "BUILD_TESTING")
+    "-DJPYLYZER_EXECUTABLE=${lib.getExe' jpylyzer "jpylyzer"}"
+    "-DOPJ_DATA_ROOT=../../data"
+    (lib.cmakeBool "BUILD_TESTING" doCheck)
```
Comment (emilazy) [inline pkgs/by-name/op/openjpeg/package.nix:57]: `BUILD_TESTING` is automatically set, no need to do it ourselves.

File: pkgs/by-name/op/openjpeg/package.nix [hunk 2]
```diff
@@ -52,12 +63,17 @@ stdenv.mkDerivation rec {
     ++ lib.optionals jpipServerSupport [ curl fcgi ]
     ++ lib.optional (jpipLibSupport) jdk;
 
-  doCheck = (!stdenv.hostPlatform.isAarch64 && !stdenv.hostPlatform.isPower64); # tests fail on aarch64-linux and powerpc64
+  # tests fail on aarch64-linux and powerpc64
```
Comment (emilazy) [inline pkgs/by-name/op/openjpeg/package.nix:66]: I can check if these pass now on `aarch64-linux` if you’d like.

**Review by emilazy — COMMENTED** [review]
Seems reasonable. Scary to disable that many tests, but it’s not like our current package was passing them anyway. (But maybe we should just disable the `jpylyzer` ones unconditionally if they all fail, if that’s a thing we can do.)

File: pkgs/by-name/op/openjpeg/package.nix (hunk 2 above)
Comment (paparodeo) [reply pkgs/by-name/op/openjpeg/package.nix:65]: thanks. I made a couple of tweaks to the `cmakeFlags` and removed `"-DCMAKE_INSTALL_NAME_DIR=\${CMAKE_INSTALL_PREFIX}/lib"` running the `passthru.tests` -- everything in `-dev` looks fine.

File: pkgs/by-name/op/openjpeg/package.nix (hunk 1 above)
Comment (paparodeo) [reply pkgs/by-name/op/openjpeg/package.nix:55]: > `BUILD_TESTING` is automatically set, no need to do it ourselves.

if I don't set it no tests run.

Comment (emilazy) [reply pkgs/by-name/op/openjpeg/package.nix:55]: Weird. I guess they flip the default or something.

**Review by emilazy — APPROVED** [review]

**Pull request merged**